lib/ansible/galaxy/api.py

```diff
@@ -332,7 +332,7 @@ def __lt__(self, other_galaxy_api):
 
         return (
             self._priority > other_galaxy_api._priority or
-            self.name < self.name
+            self.name < other_galaxy_api.name
```

Comment:
After squinting at this for a while and trying to construct a test, I don't think this is correct. I think we should delete the `or ...` instead, as this change causes inconsistent behavior. For an example, I tested installing community.aws (using --force-with-deps), with this config:
```ini
[galaxy]
server_list=server3,server1,server2

[galaxy_server.server2]
url=https://galaxy.ansible.com

[galaxy_server.server1]
url=https://console.redhat.com/api/automation-hub/
auth_url=https://sso.redhat.com/auth/realms/redhat-external/protocol/openid-connect/token
token=mytoken

[galaxy_server.server3]
url=https://galaxy.ansible.nope
```

I expect community.aws to come from Galaxy (server2) and amazon.aws to come from AH (server1).

on devel I get:
```bash
Starting collection install process
Calling Galaxy at https://galaxy.ansible.com/api/v2/collections/community/aws/versions/6.2.0/
Server https://galaxy.ansible.com/api/ has not signed community.aws:6.2.0
Fetching a collection tarball for 'community.aws:6.2.0' from Ansible Galaxy
Downloading https://galaxy.ansible.com/download/community-aws-6.2.0.tar.gz to /home/shertel/.ansible/tmp/ansible-local-2816996k1jsfj4n/tmp55ksi06a/community-aws-6.2.0-f0otruwf
Validating downloaded file hash 08c0a34ba784d1feb0c752f73105c4a88bfde8517bbf663f46a4608a0b8ba117 with expected hash 08c0a34ba784d1feb0c752f73105c4a88bfde8517bbf663f46a4608a0b8ba117
Collection 'community.aws:6.2.0' obtained from server server2 https://galaxy.ansible.com/api/
Installing 'community.aws:6.2.0' to '/home/shertel/ansible/collections/ansible_collections/community/aws'
Calling Galaxy at https://console.redhat.com/api/automation-hub/v3/collections/amazon/aws/versions/6.3.0/
community.aws:6.2.0 was installed successfully
Fetching a collection tarball for 'amazon.aws:6.3.0' from Ansible Galaxy
Downloading https://console.redhat.com/api/automation-hub/v3/plugin/ansible/content/published/collections/artifacts/amazon-aws-6.3.0.tar.gz to /home/shertel/.ansible/tmp/ansible-local-2816996k1jsfj4n/tmp55ksi06a/amazon-aws-6.3.0-jb7_hadd
Validating downloaded file hash 6fe1da14018b7b4ab8c995263bb41219303fc33dd6c5bd1c3b2fd7b65b23929c with expected hash 6fe1da14018b7b4ab8c995263bb41219303fc33dd6c5bd1c3b2fd7b65b23929c
Collection 'amazon.aws:6.3.0' obtained from server server1 https://console.redhat.com/api/automation-hub/
Installing 'amazon.aws:6.3.0' to '/home/shertel/ansible/collections/ansible_collections/amazon/aws'
amazon.aws:6.3.0 was installed successfully
```

on this branch I can get:
```bash
Starting collection install process
Calling Galaxy at https://galaxy.ansible.com/api/v2/collections/community/aws/versions/6.2.0/
Server https://galaxy.ansible.com/api/ has not signed community.aws:6.2.0
Fetching a collection tarball for 'community.aws:6.2.0' from Ansible Galaxy
Downloading https://galaxy.ansible.com/download/community-aws-6.2.0.tar.gz to /home/shertel/.ansible/tmp/ansible-local-2817098p6d5jowa/tmpo58ue24w/community-aws-6.2.0-_u2dyn6h
Validating downloaded file hash 08c0a34ba784d1feb0c752f73105c4a88bfde8517bbf663f46a4608a0b8ba117 with expected hash 08c0a34ba784d1feb0c752f73105c4a88bfde8517bbf663f46a4608a0b8ba117
Collection 'community.aws:6.2.0' obtained from server server2 https://galaxy.ansible.com/api/
Installing 'community.aws:6.2.0' to '/home/shertel/ansible/collections/ansible_collections/community/aws'
Calling Galaxy at https://galaxy.ansible.com/api/v2/collections/amazon/aws/versions/6.3.0/
community.aws:6.2.0 was installed successfully
Server https://galaxy.ansible.com/api/ has not signed amazon.aws:6.3.0
Fetching a collection tarball for 'amazon.aws:6.3.0' from Ansible Galaxy
Downloading https://galaxy.ansible.com/download/amazon-aws-6.3.0.tar.gz to /home/shertel/.ansible/tmp/ansible-local-2817098p6d5jowa/tmpo58ue24w/amazon-aws-6.3.0-ux5kad7v
Validating downloaded file hash bd69f0f730401263ad0e2065ae3f40bb58873fd261a4cace70e24bd6b9bad4a9 with expected hash bd69f0f730401263ad0e2065ae3f40bb58873fd261a4cace70e24bd6b9bad4a9
Collection 'amazon.aws:6.3.0' obtained from server server2 https://galaxy.ansible.com/api/
Installing 'amazon.aws:6.3.0' to '/home/shertel/ansible/collections/ansible_collections/amazon/aws'
amazon.aws:6.3.0 was installed successfully
```

Although this result isn't consistent (it depends on which servers are being compared), on devel `server1` is never less than `server2`. On this branch, `server1` is always less than `server2`, because now the alphabetical order matters. I think it's a bug to consider the server names unless the priorities are equal, but not sure if this can happen.

Reply:
Did _I_ write that? What was I thinking...

Reply:
@webknjaz Maybe it was a preventative measure against priority collisions initially (a `source` for an individual requirement gets a priority of infinity), but it doesn't appear possible that we ever compare two servers with inf priority. The requirement with the inf priority source uses the source, and dependencies just use the configured servers list.

If the search behavior for dependencies changes to also search the parent collection's source for the dependency, we might need to add a `or (self._priority == other_galaxy_api._priority and self.name < other_galaxy_api.name)` here to ensure a consistent sort order for latest_matches.